### PR TITLE
RUN-1407: reduce unnecessary getObjectPreview work

### PR DESCRIPTION
**Pull request by Domiii**
* https://linear.app/replay/issue/RUN-1407/improve-getobjectpreview-performance-part-1

### CODE REVIEW

File: third_party/blink/renderer/bindings/core/v8/record_replay_interface.cc [hunk 1]
```diff
@@ -276,9 +276,15 @@ function commandCallback(method, params) {
...
   try {
+    // track performance temporarily (RUN-1315)
+    const startTime = Date.now();
     VerboseCommands && log(`[Command ${method}] Handling command, params=${JSON_stringify(params)}...`);
     const result = CommandCallbacks[method](params);
     VerboseCommands && log(`[Command ${method}] Handled command, result=${JSON_stringify(result)}`);
+    const timeTaken = (Date.now() - startTime)/1000;
+    if (timeTaken > 0.2) {
+      log(`[RuntimeWarning][Command ${method}] slow: ${timeTaken.toFixed(2)}s, params: ${JSON.stringify(params)}`);
+    }
```
Comment (bhackett1024) [inline third_party/blink/renderer/bindings/core/v8/record_replay_interface.cc:287]: This should be removed, the recorder also tracks the time taken and we can get the information from those logs.  In particular we can't log the params as they can contain sensitive user information.

File: third_party/blink/renderer/bindings/core/v8/record_replay_interface.cc [hunk 2]
```diff
@@ -478,11 +484,18 @@ function Pause_evaluateInGlobal({ expression }) {
...
   const frames = getStackFrames().map((frame, index) => {
     // Use our own IDs for frames.
     const id = (index++).toString();
     return createProtocolFrame(id, frame);
   });
+  
+  const timeTaken = (Date.now() - startTime)/1000;
+  if (timeTaken > 0.2) {
+    log(`[RuntimeWarning] Pause_getAllFrames slow: ${timeTaken.toFixed(2)}s`);
+  }
```
Comment (bhackett1024) [inline third_party/blink/renderer/bindings/core/v8/record_replay_interface.cc:498]: ditto

File: third_party/blink/renderer/bindings/core/v8/record_replay_interface.cc [hunk 3]
```diff
@@ -1084,11 +1099,19 @@ ProtocolObjectPreview.prototype = {
...
     const cdpProperties = sendMessage("Runtime.getProperties", {
       objectId: this.cdpObj.objectId,
       ownProperties: true,
       generatePreview: false,
     });
+    const timeTaken = (Date.now() - startTime)/1000;
+    if (timeTaken > 0.2) {
+      const len = Object.keys(this.raw).length;
+      log(`[RuntimeWarning] Runtime.getProperties (${this.rrpId}) slow: ${timeTaken.toFixed(2)}s (${len})`);
+    }
```
Comment (bhackett1024) [inline third_party/blink/renderer/bindings/core/v8/record_replay_interface.cc:1114]: ditto

**Review by bhackett1024 — APPROVED** [review]

File: third_party/blink/renderer/bindings/core/v8/record_replay_interface.cc (hunk 1 above)
Comment (Domiii) [reply third_party/blink/renderer/bindings/core/v8/record_replay_interface.cc:287]: Done x3

**Pull request merged**